include/MySQL_Protocol.h

```diff
@@ -6,8 +6,16 @@
 
 #define RESULTSET_BUFLEN 16300
 
+/* The default mariadb-connecter 3.1.4 does not yet implement CLIENT_DEPRECATE_EOF
+ * flag.
+ */
+#ifndef CLIENT_DEPRECATE_EOF
+#define CLIENT_DEPRECATE_EOF     (1UL << 24)
+#endif
```

Comment:
Mariadb client support PR is [here](https://github.com/mariadb-corporation/mariadb-connector-c). Until that gets merged, this will make the code compile properly.